BodoSQL/calcite_sql/bodosql-calcite-application/src/main/java/com/bodosql/calcite/rel/metadata/BodoRelMdColumnUniqueness.kt

```diff
@@ -0,0 +1,357 @@
+package com.bodosql.calcite.rel.metadata
+
+import org.apache.calcite.plan.RelOptPredicateList
+import org.apache.calcite.rel.RelNode
+import org.apache.calcite.rel.RelShuttleImpl
+import org.apache.calcite.rel.core.Join
+import org.apache.calcite.rel.core.JoinRelType
+import org.apache.calcite.rel.logical.LogicalFilter
+import org.apache.calcite.rel.metadata.BuiltInMetadata.ColumnUniqueness
+import org.apache.calcite.rel.metadata.MetadataDef
+import org.apache.calcite.rel.metadata.MetadataHandler
+import org.apache.calcite.rel.metadata.RelMetadataQuery
+import org.apache.calcite.rex.RexCall
+import org.apache.calcite.rex.RexInputRef
+import org.apache.calcite.rex.RexNode
+import org.apache.calcite.rex.RexSubQuery
+import org.apache.calcite.rex.RexUtil
+import org.apache.calcite.sql.SqlKind
+import org.apache.calcite.util.ImmutableBitSet
+import org.apache.calcite.util.Pair
+import org.apache.commons.lang3.mutable.MutableBoolean
+import java.util.BitSet
+import java.util.function.Consumer
+
+class BodoRelMdColumnUniqueness : MetadataHandler<ColumnUniqueness> {
+    override fun getDef(): MetadataDef<ColumnUniqueness> = ColumnUniqueness.DEF
+
+    fun areColumnsUnique(
```

Comment:
This is the only function we modified. The only change is that I had the IDE automatically convert it into Kotlin.